src/stake.rs

```diff
@@ -0,0 +1,103 @@
+use crate::{Error, Height};
+
+use fuel_crypto::PublicKey;
+use fuel_types::Bytes64;
+use hashbrown::HashMap;
+
+use core::ops::{Range, RangeBounds};
+
+mod stake_keys;
+
+use stake_keys::StakeKeys;
+
+/// Registered stake for a validator
+///
+/// The used key might not reflect the canonical validators set and will be used only to verify the
+/// signatures with the chosen protocol.
+#[derive(Default, Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Stake {
+    /// One-time key for a height range
```

Comment:
[nit] hot key would sound a little more accurate, as one time keys are generally only used once (like in btc).